### DISCUSSION TITLE
proxy support implementation.

### DISCUSSION THREAD

**Discussion by ghost** [rustdesk/rustdesk#831] [General]
I suggest you add a support for proxies other than socks5. Would be useful for corporate usage and self-hosting. Where you can choose port and address.

**Comment by ghost** [comment]
In my case, it would be a squid proxy compatibility

**Reply by gstorelli** [reply]
I was writing a similar request. You already enabled Socks5 Proxy. I just need simple HTTP Proxy with authentication.

**Reply by mdevil666** [reply]
would need http proxy support for a squid proxy too

**Reply by RafaelLinux** [reply]
Me too. I just wrote a thread (that I'll try to delete) suggesting to add HTTP(S) proxy option. I can't use RustDesk without that option.

**Comment by mcarbonneaux** [comment]
in majority of corporate, the internet access are accessible by this way (socks5 is too open for us) with http proxy....

http proxy use CONNECT method to open tunnel, but has limitation, they cannot be used for udp protocol only tcp....

generaly is to send http request CONNECT with the "host:port" (with proxy authorization if auth is needed) of the destination, and after leave the application to communicate...

**Comment by mcarbonneaux** [comment]
https://crates.io/crates/async-http-proxy

**Comment by AlexanderGabriel** [comment]
We do not allow any socket connections to the internet in our company network.
Everything is through proxy.
We f.e. have jitsi working like this. it uses websocket an reverse proxy to communicate with the videobridges when clients are behind a proxy.
our proxy has basic authentication so the rustdesk client should be able to ask for a password.
discovered rustdesk yesterday and it is really awesome! would like to test this at work but without http proxy support... no way...
i know this is not so easy because http proxy is a different protocol and not just forwarding packets like SOCKS and i'm really sorry i can not write rust but if you implement this, i would be happy to help with test in our environment and give feedback

**Comment by oumad** [comment]
We have this need too, I just discovered RustDesk and I wonder why I didn't know about it much earlier, in the environment I need to use it I have to indicate an http_proxy (and https_proxy) environment variable containing something like `http://[user]:[password]@[host]:[port]` for things like npm and git to work, anydesk and teamviewer have a UI that let's you input that.

**Comment by cyfrax** [comment]
this would be great. I need to remote control a PC behind corporate proxy. Anydesk works, rustdesk not. Would love to see rustdesk supporting this.

**Comment by rustdesk** [comment]
Close as finished, https://twitter.com/rustdesk/status/1783468475480191022
